### PR TITLE
made a guided project on bmi-calculator with steps and challenges

### PR DESCRIPTION
**Pull request by pranayjalan254**
made a repository where the design and the html boilerplate code is ready. Please review it

### CODE REVIEW

**Review by Pranavtechie — CHANGES_REQUESTED** [review]
@pranayjalan254 Can you update the step folder names. 

The folder names should not consist of `space` 

The `stepId` should match with the folder name. 

So the folder name should be `step1` , `step2` ...

**Comment by pranayjalan254** [comment]
yeah sure.

**Comment by pranayjalan254** [comment]
if you could help me how to change the folder names, it would be great help.

**Comment by pranayjalan254** [comment]
Kindly review it. Please. Will modify it as required

**Comment by Pranavtechie** [comment]
@pranayjalan254  Your boilerplate github repo does not contain the `.cdmrc` file. Can you add it. You can refer to this [documentation](https://codedamn.com/news/product/how-to-use-codedamn-playgrounds#File_READMEmd). 

Adding the below mentioned lines is enough for now. 

```
terminal-one: static-server -p 1337 --no-cache
tabs: ['index.html']
```

**Comment by pranayjalan254** [comment]
done sir, kindly review and merge my pull request.

**Comment by pranayjalan254** [comment]
@mehulmpt @Pranavtechie please review this pull request and ask for changes if required..

File: bmi-calculator/spec.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+name: Body Mass Index Calculator
+tags: ["HTML/CSS", "JavaScript"]
+starter-files: https://github.com/pranayjalan254/codedamn-bmi-calculator
+type: frontend
+level: easy
+cover-image: https://github.com/pranayjalan254/codedamn-bmi-calculator/blob/main/designs/Final%20design.jpg
```
Comment (Pranavtechie) [inline bmi-calculator/spec.yml:7]: Can you update the cover-image so that it uses the direct image link : the link would be

https://raw.githubusercontent.com/pranayjalan254/codedamn-bmi-calculator/main/designs/Final%20design.jpg

for your image.

**Comment by pranayjalan254** [comment]
Updated the cover image link. Please review it

File: bmi-calculator/step0/Instructions.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+A healthy and normal BMI should be between 18.5 and 25. It is underweight if its value is less than 18, which is unhealthy. It is considered fat or overweight if its value exceeds 25.
+
+There are some criteria that will help you determine which class you belong in, such as underweight, healthy, overweight, and obese. You can better understand it by looking at the chart below.
```
Comment (Pranavtechie) [inline bmi-calculator/step0/Instructions.md:3]: You can also include elaborate instructions here (not necessarily complete code) that help in solving the challenges mentioned in each steps

File: bmi-calculator/step1/Instructions.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## In this step you need to add the heading and slider using HTML.
+
+## Style it as you want
```
Comment (Pranavtechie) [inline bmi-calculator/step1/Instructions.md:3]: You could also have mentioned the current styles that you have used here

**Review by Pranavtechie — CHANGES_REQUESTED** [review]
Try and improve the instructions given in all steps, the instructions should be as elaborate as possible, such that they help the user in completing the given challenges in each steps. 

You can refer to the already built guided projects, bird-app and multiverse-photo-gallery-project

**Comment by pranayjalan254** [comment]
okay sir, will do the required changes

**Comment by pranayjalan254** [comment]
Done. Kindly review it and all the typos (if there) are fixed. @Pranavtechie

File: bmi-calculator/step3/Instructions.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+## In this step you need to create a view of the calculator. The webpage should be styled accordingly.
+
+## Use some styles to add the background color and some width and margin to the box. Also add some styles like increasing font size and text-alignment property to the webpage
```
Comment (Pranavtechie) [inline bmi-calculator/step3/Instructions.md:3]: You could also mention the styles that the user has to use, at least the width and typography

File: bmi-calculator/step3/challenges.yml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+name: 'Create a display to view BMI'
+stepId: step3
+stepBreakDown:
+    - text: 'Create a display that will show calculations'
+    - text: 'No set height and will be determined by the amount of content'
```
Comment (Pranavtechie) [inline bmi-calculator/step3/challenges.yml:5]: Can you rephrase these two challenges, I couldn't understand them correctly. Let me know what you were trying to convey

File: bmi-calculator/step4/challenges.yml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+name: 'Add the functionality to our calculator'
+stepId: step4
+stepBreakDown:
+    - text: 'Define variables for range sliders and weight variable'
+    - text: 'display the  slider values using .innerHTML and .value property'
+    - text: 'add the formula for calculating the BMi. First and foremost, we will accept the user’s weight and height, and then we will add that value to the formula. Whatever the value is, we will compare it using the for loop, and then define the person as underweight, healthy, overweight, or obese based on that value.'
```
Comment (Pranavtechie) [inline bmi-calculator/step4/challenges.yml:6]: Here, I can see the role reversal of challenges and instructions page. The step4 breakdown can be re-written. The challenges need to concise & to the point, whereas the instructions should give an elaborate explanation of what needs to be done with some details.

**Review by Pranavtechie — CHANGES_REQUESTED** [review]

File: bmi-calculator/step2/challenges.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+name: 'Add the calculate using HTML'
+stepId: step2
+stepBreakDown:
+    - text: 'Now using the input with type as “button” add the onlick function to our button for the javascript button . Also use some style to add the background color and with some width and margin to our input box .'
```
[the remark this answers is not in the document]
Comment (pranayjalan254) [reply bmi-calculator/step2/challenges.yml:4]: i did not get you, please elaborate

**Comment by pranayjalan254** [comment]
Please review. @Pranavtechie

**Comment by pranayjalan254** [comment]
Please tell if any other things to be modified... Thank You

**Comment by pranayjalan254** [comment]
@Pranavtechie

**Comment by pranayjalan254** [comment]
hey @Pranavtechie did received any further reviews for my project...Is the PR completed

**Comment by pranayjalan254** [comment]
No further comments?? @Pranavtechie @mehulmpt

**Comment by mehulmpt** [comment]
Hi Pranay
Continuously pinging us to review would not help. If anything, it just annoys me and everyone who subscribed to this issue. We are not ignoring our PR, we will review it. Please be patient.

File: bmi-calculator/spec.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+
+name: Body Mass Index Calculator
+tags: ["HTML/CSS", "JavaScript"]
+starter-files: https://github.com/pranayjalan254/codedamn-bmi-calculator
+type: frontend
+level: easy
+cover-image: https://raw.githubusercontent.com/pranayjalan254/codedamn-bmi-calculator/main/designs/Final%20design.jpg
+short-description: A simple bmi calculator usin HTML, CSS and some Javascript.
+long-description: |
+  Your challenge is to build out this bmi calculator and get it looking as close to the design as possible.
```
Comment (mehulmpt) [inline bmi-calculator/spec.yml:10]: Your `long-description` field is completely wrong (and copied from other projects). Please refactor this to include proper changes.

File: bmi-calculator/step1/Instructions.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+
+Import Roboto font from Google Fonts
+
+background-color: #f6f6f6
```
Comment (mehulmpt) [inline bmi-calculator/step1/Instructions.md:7]: What is this exactly? Code or content copy? If code, please use backticks and syntax highlighting

File: bmi-calculator/step1/Instructions.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
+## In this step you need to add the heading and slider using HTML.
```
Comment (mehulmpt) [inline bmi-calculator/step1/Instructions.md:1]: Why do you have two headings continuously?

File: bmi-calculator/step2/Instructions.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+## In this step you need to add the buttons for your bmi calculator.
+
+## Make sure you style it properly
```
Comment (mehulmpt) [inline bmi-calculator/step2/Instructions.md:3]: Wrong use of markdown headings

File: bmi-calculator/step4/Instructions.md [hunk 12]
```diff
@@ -0,0 +1,15 @@
+## This is the main step.
+
+## Add functionality to the calculator using javascript.
+
+## Define the variables and properties as required by following the steps
+
+## Add the bmi formula using javascript.
+
+## First and foremost, we will accept the user’s weight and height, and then we will add that value to the formula. Whatever the value is, we will compare it using the for loop, and then define the person as underweight, healthy, overweight, or obese based on that value.
```
Comment (mehulmpt) [inline bmi-calculator/step4/Instructions.md:9]: Wrong use of markdown headings

Comment (mehulmpt) [reply bmi-calculator/step4/Instructions.md:9]: Please learn about markdown: https://www.markdownguide.org/basic-syntax/

**Review by mehulmpt — CHANGES_REQUESTED** [review]
Need many changes

**Comment by pranayjalan254** [comment]
> Hi Pranay Continuously pinging us to review would not help. If anything, it just annoys me and everyone who subscribed to this issue. We are not ignoring our PR, we will review it. Please be patient.

I am extremely sorry sir, will do the required changes. Will not ping you again. Sorry

**Review by Pranavtechie — CHANGES_REQUESTED** [review]
Hey @pranayjalan254 , please go through the written text before submitting, there are lot of simple mistakes that could have been avoided. 

Before starting to edit this project, I'd recommend you to get the complete idea of how our guided project work. Please go through an existing guide projects code and try and open it as a guided project on the website to understand it's working. 

This will give you a good idea on how to write the project.

**Comment by Pranavtechie** [comment]
@pranayjalan254 are you working on the requested changes, just wanted to get an update.

**Comment by pranayjalan254** [comment]
sorry, was busy settling in the admitted college, will try to resume the work soon

**Comment by Pranavtechie** [comment]
@pranayjalan254 Please make the requested changes as soon as possible

**Comment by Pranavtechie** [comment]
@pranayjalan254 The challenges and the Instructions are not up to the mark and would require a complete re-write from your end. I'm closing this PR, feel free to open a new PR when you have rewritten the challenges. 

Please refer to existing projects to understand how to write the challenges and Instructions. 

As this is a beginner project the instructions need to be very clear and explain why you are taking the particular step. You can refer to https://codedamn.com/project/multiverse-html5-photo-gallery/
